sphinx_immaterial/graphviz.py

```diff
@@ -235,10 +235,16 @@ def render_dot_html(
 ) -> Tuple[str, str]:
 
     theme_options = self.builder.config["html_theme_options"]
-    ttf_font_paths = google_fonts.get_ttf_font_paths(self.builder.app)
-    font = theme_options["font"]["text"]
+    if theme_options["font"]:
+        font = theme_options["font"]["text"]
+    else:
+        font = "Helvetica"
```

Comment:
If we allow `html_theme_options["font"]` to be used even without injecting google fonts, then we can have graphviz still use the system or user-defined font in its output, just with the metrics of Helvetica.  I think that would be better than always actually displaying Helvetica if the user disables google fonts.

Reply:
I did this to avoid
```
>       font = theme_options["font"]["text"]
E       TypeError: 'bool' object is not subscriptable
```
when font is false. This could be handled better though.
```py
    font: str = "Helvetica"
    if isinstance(theme_options["font"], dict) and "text" in theme_options["font"]:
        font = theme_options["font"]["text"]
```
Or did you have something else in mind?

Reply:
> then we can have graphviz still use the system or user-defined font in its output, just with the metrics of Helvetica

My main confusion is here. Sorry this post got real long...

The user doesn't need to explicitly specify a system font name. Rather, `font = False` will use whatever the theme's CSS var has set for system-specific fallbacks.
https://github.com/jbms/sphinx-immaterial/blob/cf059973255ee2595005765b941c81b47bbcc380/src/assets/stylesheets/main/_typeset.scss#L32-L35
Because **build system's offerings can differ from client system's offerings**, the exact system font used during browsing is unknown at build time when `font = False`.

Furthermore, as I mentioned before, explicitly setting a system font name would be done in user-defined CSS:
```css
:root {
  --md-text-font: "Arial";
}
```
Note, this var isn't required to be in `:root {` and can be specified more than once.
Because `font` is still `False` in this scenario, we have no easy way of telling graphviz which system font to use at build time. We could crawl the user-defined CSS for this var, but that would lead to a guessing game if more than 1 `--md-text-font` var is specified under different CSS selectors.

---
I suppose I could exclude the `-<x>fontname={ttf_font}` values
https://github.com/jbms/sphinx-immaterial/blob/cf059973255ee2595005765b941c81b47bbcc380/sphinx_immaterial/graphviz.py#L274-L283
when no google fonts are used or a system font is implied via `font = False`. But, I think that incites graphviz to use Helvetica (or Arial) by default anyway. I'm hoping this is the intentionally requested change because I'm rather confused.

Reply:
Now that the CI was working through the linting/format checks, I see how my original approach only worked on Windows. So, I ended up only adding the `-<x>fontname={ttf_font}` values if `ttf_font` is not None, and prevented modifying the SVG elements' `font-family` attribute when `font` is None.

The unit test output renders fine: 
![image](https://user-images.githubusercontent.com/14963867/221840909-0a3b117e-2526-4631-96d6-f30d3dbc2f17.png)
And the CI seems happy now.

Reply:
Note that there are really two fonts used for the graphviz extension: the font whose metrics are used by graphviz itself to place things appropriately in the svg output ("metrics font"), and the font that actually gets displayed in the browser ("display font").  Ideally these two fonts are the same, but even if we just use the metrics from a default font like Helvetica, it only makes a small difference and isn't a big deal.

I agree that if the user is just setting the font manually via CSS, then we don't have a robust way of knowing what the font will be at build time and therefore have no hope of specifying a "metrics font".  Additionally, even if we did know what the font will be, we wouldn't be able to tell graphviz how to load it anyway.  I think we could still use the custom font as the "display font", though, since we could just set the font to `var(--md-text-font-family)`.  I think that would be better than just using the default graphviz font.

Reply:
> we could just set the font to `var(--md-text-font-family)`

👀 I'll give that a try. It might even resolve the system font fallback used in CSS (for browsers only, not font metrics obviously).

Reply:
Ok adding `var(--md-text-font-family)` to the inline style of SVG nodes that have the font-family attr set works to render the graph with the system font (if not using a cached google font):
![image](https://user-images.githubusercontent.com/14963867/222286367-df67160b-f2cc-4361-9c3b-1397240196b3.png)
Notice the graphviz default  font metrics are still used. In the above rendering, the graph nodes' `text` elements use `font-family = "Times New Roman"`, but the inline `style="font-family: var(--md-text-font-family);"` overrides the text elements' attr.
```xml
<text text-anchor="middle" x="27" y="-86.9" font-family="Times New Roman,serif" font-size="12.00"
  style="fill: var(--md-code-fg-color); font-family: var(--md-text-font-family)">A</text>
```

Reply:
I think in case that we are setting `font-family` in the style we should also remove the `font-family` attribute.

Reply:
done.